R/guides-.r

```diff
@@ -190,10 +197,12 @@ guides_train <- function(scales, theme, guides, labels) {
       #   + guides(XXX) > + scale_ZZZ(guide=XXX) > default(i.e., legend)
       guide <- resolve_guide(output, scale, guides)
 
-      # this should be changed to testing guide == "none"
-      # scale$legend is backward compatibility
-      # if guides(XXX=FALSE), then scale_ZZZ(guides=XXX) is discarded.
-      if (identical(guide, "none") || isFALSE(guide) || inherits(guide, "guide_none")) next
+      if (identical(guide, "none") || inherits(guide, "guide_none")) next
+
+      if (isFALSE(guide)) {
+        warn('It is deprecated to specify `FALSE` to remove a guide. Please use `"none"` instead.')
```

Comment:
Is there a realistic usage pattern by which we would reach this branch? What happens when somebody adds `scale_*(guide = FALSE)`? Is it caught here or in the other check?

Reply:
> `scale_*(guide = FALSE)`

Ah, yes. It reaches here. Maybe I should warn it on `(discrete|continuous|binned)_scale()`? What I was in mind is that there might be some extension package that creates guide object without `guides()`. Not sure if there's any.

Reply:
Why not just write "It is deprecated to specify `guide = FALSE` to remove a guide. Please use `guide = "none"` instead."?

Reply:
I see what you mean. Makes sense, thanks.